models/migrations/v128.go

```diff
@@ -83,17 +83,17 @@ func fixMergeBase(x *xorm.Engine) error {
 
 			if !pr.HasMerged {
 				var err error
-				pr.MergeBase, _, err = git.NewCommand(git.DefaultContext, "merge-base", "--", pr.BaseBranch, gitRefName).RunStdString(&git.RunOpts{Dir: repoPath})
+				pr.MergeBase, _, err = git.NewCommand(git.DefaultContext, "merge-base").AddSlashedArguments(pr.BaseBranch, gitRefName).RunStdString(&git.RunOpts{Dir: repoPath})
 				if err != nil {
 					var err2 error
-					pr.MergeBase, _, err2 = git.NewCommand(git.DefaultContext, "rev-parse", git.BranchPrefix+pr.BaseBranch).RunStdString(&git.RunOpts{Dir: repoPath})
+					pr.MergeBase, _, err2 = git.NewCommand(git.DefaultContext, "rev-parse").AddDynamicArguments(git.BranchPrefix + pr.BaseBranch).RunStdString(&git.RunOpts{Dir: repoPath})
```

Comment:
Maybe a bit late to the party but for me the names `AddSlashedArguments` and `AddDynamicArguments` are not very descriptive. In the example above `pr.BaseBranch` is added one time with `AddSlashedArguments` and the other time with `AddDynamicArguments`. Without deep understanding what both methods do I can't tell the difference.

How about renaming `AddSlashedArguments` to `ForceEndOfOptionsAndAddArguments` (🎉) and `AddDynamicArguments` to `AddArgumentsAndPreventOptions` or `AddUnsafeArguments`. Feel free to propose better names.

Reply:
Because the code base is messy at the moment. Many arguments are not added by proper methods.

This PR's purpose is just making everything safe, but not changing any existing logic.

Actually I have tried `AddUnsafeArguments` before, but it's even more unclear .... because many dynamic arguments (user-provided or constructed by code)  are safe.


`ForceEndOfOptionsAndAddArguments` doesn't work either because there are two different options: `--end-of-options` and `--`, they are not the same.

Reply:
And you could even see other mistakes in old code, like this:

![image](https://user-images.githubusercontent.com/2114189/197390656-f7bd8873-6b3a-4dbf-a8db-49b209a86dbc.png)

I am sure the old `--` was at the wrong place, but as a (big) refactoring, I do not want to touch the old logic. This PR is safe enough and should work well in all cases.

Leave more improvements to next PRs.

Reply:
> Because the code base is messy at the moment. Many arguments are not added by proper methods.
> 
> This PR's purpose is just making everything safe, but not changing any existing logic.

Changing the method names does not change existing logic.

> Actually I have tried `AddUnsafeArguments` before, but it's even more unclear .... because many dynamic arguments (user-provided or constructed by code) are safe.

How about `AddPossiblyUnsafeArguments`?

> `ForceEndOfOptionsAndAddArguments` doesn't work either because there are two different options: `--end-of-options` and `--`, they are not the same.

We don't use `--end-of-options` at the moment. GitLab calls arguments after `--` "post separator args".

And why `Slashed` if the method adds `--`? Those are `dashes` and not `slashes` (`/`).

Reply:
Oh yes .... I chosen the wrong word ....... then `AddDashedArguments`?

Reply:
To get this PR in, yes.

Reply:
> Changing the method names does not change existing logic.

I am answering "why the different methods are used to BaseBranch"  😂

> How about AddPossiblyUnsafeArguments?

It seems too wordy .....

> And why Slashed if the method adds --? Those are dashes and not slashes (/).

Oh yes .... I chosen the wrong word ....... then AddDashedArguments?